atomicapp/providers/docker.py

```diff
@@ -54,37 +55,49 @@ def init(self):
             logger.info("DRY-RUN: Did not check Docker version compatibility")
         else:
             cmd_check = ["docker", "version"]
-            try:
-                docker_version = subprocess.check_output(cmd_check).split("\n")
-            except Exception as ex:
-                raise ProviderFailedException(ex)
-
-            client = ""
-            server = ""
-            for line in docker_version:
-                if line.startswith("Client API version"):
-                    client = line.split(":")[1]
-                if line.startswith("Server API version"):
-                    server = line.split(":")[1]
-
-            if client > server:
-                msg = ("Docker version in app image (%s) is higher than the one "
-                       "on host (%s). Please update your host." % (client, server))
-                raise ProviderFailedException(msg)
+
+            version = self.shipy(cmd_check)
+
+            compatibility = {
+                '1.21': ('1.6.0', '1.7.0', '1.7.1', '1.7.2'),
+                '1.22': ('1.8.0', '1.8.1'),
+                '1.23': ('1.8.0', '1.8.1')
+            }
+
+            if version.dpy not in compatibility[version.sapi]:
+                raise Exception('docker-py version {}, using API version {} '
+                                'is not compatible with docker server using '
+                                'API version {}.\n'
+                                'Please install docker-py version one of '
+                                'the following {}'.format(
+                                    version.dpy,
+                                    version.capi,
+                                    version.sapi,
+                                    compatibility[version.sapi]))
+
+    def shipy(self, cmd):
+        return self.shipy_exec.shipy(cmd[1:], external_logger=logger)
 
     def _get_containers(self):
-        docker_cmd = 'docker ps -a --format="{{ .Names }}"'
+        docker_cmd = 'docker ps -a'
         if self.dryrun:
             logger.info("DRY-RUN: %s", docker_cmd)
             return []
         else:
-            return dict((line, 1) for line in subprocess.check_output(docker_cmd, shell=True).splitlines())
+            containers = []
+
+            output = self.shipy(docker_cmd.split())
+            for container in output:
```

Comment:
can we do `for line in output` instead. It's just a string of output correct? or is it more than that?